Lesson_Materials/In_Order_Queue/index.html

```diff
@@ -155,34 +151,53 @@
 					<div class="container">
 						<div class="col">
 							<code class="code-100pc"><pre>
-auto devicePtr   = usm_wrapper&lt;int&gt;(
-  malloc_device&lt;int&gt;(1024, inOrderQueue));
-  
-  inOrderQueue.memcpy(devicePtr, data, sizeof(int));
-  
-  inOrderQueue.parallel_for&lt;kernel_a&gt;(sycl::range{1024},
-  [=](sycl::id&lt;1&gt; idx){
-  devicePtr[idx] = /* some computation */
-});
-
-inOrderQueue.parallel_for&lt;kernel_b&gt;(sycl::range{1024},
-  [=](sycl::id&lt;1&gt; idx){
-  devicePtr[idx] = /* some computation */
-});
-
-inOrderQueue.memcpy(data, devicePtr, sizeof(int));
+sycl::queue Q{sycl::property::queue::in_order};
+for (std::function task: tasks)
+  Q.submit(task);
+Q.wait();
+							</code></pre>
+						</div>
+						<div class="col" data-markdown>
+							* All commands are executed serially
+							* Ease of programming (no race conditions can occur)
+							* Potentialy lower-latency than out-of-order queues
+							* Zero concurrency / Potentially suboptimal performance
+						</div>
+					</div>
+				</section>
+				<section>
+					<div class="hbox" data-markdown>
+						#### Using an out-of-order queue to enable concurrency in kernel execution (USM)
+					</div>
+					<div class="container">
+						<div class="col">
+							<code class="code-100pc"><pre>
+std::vector<std::function, sycl::queue> queues_tasks;
```

Comment:
```suggestion
std::vector<std::pair<std::function, sycl::queue>> queues_tasks;
```

Reply:
Good catch, this is the problem with pseudo code... No compiler!